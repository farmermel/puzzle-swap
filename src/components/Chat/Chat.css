.chat {
  margin: 10px 0;
  background-color: #fdfdfd;
  height: 90px;
  border: 1px solid #eaeaea;
  box-shadow: 0 1px 2px #999;
  cursor: pointer;
  transition: .25s;
  width: 80%;
  max-width: 700px; }
  .chat:hover {
    box-shadow: 0 1px 20px #999; }

.chat a {
  text-decoration: none;
  display: flex;
  align-items: center;
  height: 100%;
  color: #000;
  justify-content: space-between;
  padding: 3px 40px 20px 40px; }

.chat-member {
  font-weight: 300;
  text-align: left;
  width: 80px; }

.last-message {
  text-align: left;
  text-overflow: ellipsis;
  overflow: hidden;
  width: 360px;
  white-space: nowrap; }

.last-message p,
.last-message h2 {
  line-height: 30px;
  margin: 0; }

.last-message h2 {
  color: #daa100;
  font-weight: 800; }

.chat-timestamp {
  font-weight: 200;
  font-size: .9em;
  text-align: right;
  width: 100px; }
