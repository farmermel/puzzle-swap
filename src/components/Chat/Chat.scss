.chat {
  margin: 10px 0;
  background-color: #fdfdfd;
  height: 90px;
  border: 1px solid #eaeaea;
  box-shadow: 0 1px 2px #999;
  cursor: pointer;
  transition: .25s;
  width: 80%;
  max-width: 700px;
  &:hover {
    box-shadow: 0 1px 20px #999;
  }
}

.chat a {
  text-decoration: none;
  display: flex;
  align-items: center;
  height: 100%;
  color: #000;
  justify-content: space-between;
  padding: 3px 40px 20px 40px;
}

.last-message {
  text-overflow: ellipsis;
  width: 360px;
  overflow: hidden;
  white-space: nowrap;
}

.chat-timestamp {
  text-align: right;
  width: 100px;
}