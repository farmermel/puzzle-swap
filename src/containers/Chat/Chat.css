.chat {
  background-color: #fdfdfd;
  height: 210px;
  width: 80%;
}