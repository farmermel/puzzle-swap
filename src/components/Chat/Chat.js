import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Chat.css';

const getMembers = (members, userId) => {
  const messager = Object.keys(members).find( member => {
    return member !== userId && members[member].username;
  })
  return members[messager].username;
}

const Chat = ({ chat, userId }) => {
  return (
    // <div className='hover-wrap'>
      <div className='chat'>
        <Link to={{ pathname: `/messages/${chat.chatId}`}}>
          <p>{getMembers(chat.members, userId)}</p>
          <p className='last-message'>{chat.lastMessage}</p>
          <p className='chat-timestamp'>{chat.timeStamp}</p>
        </Link>
      </div>
    // </div>
  )
}

Chat.propTypes = {
  chat: PropTypes.shape({
    timeStamp: PropTypes.string,
    members: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
    chatId: PropTypes.string,
    lastMessage: PropTypes.string
  })
}

export default Chat;