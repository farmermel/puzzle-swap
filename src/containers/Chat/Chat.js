import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Chat.css';

const getMembers = members => {
  return Object.keys(members).map( member => (
    members[member].username
  ))
}

const Chat = ({ chat }) => {
  return (
    <div className='chat'>
      <Link to={{ pathname: `/messages/${chat.chatId}`}}>
        <h3>Members: <span>{getMembers(chat.members)}</span></h3>
        <h2>{chat.timeStamp}</h2>
        <p>{chat.lastMessage}</p>
      </Link>
    </div>
  )
}

Chat.propTypes = {
  chat: PropTypes.shape({
    timeStamp: PropTypes.number,
    members: PropTypes.objectOf(PropTypes.string),
    chatId: PropTypes.string,
    lastMessage: PropTypes.string
  })
}

export default Chat;