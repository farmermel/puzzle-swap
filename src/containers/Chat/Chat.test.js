import React from 'react';
import { shallow } from 'enzyme';
import Chat from './Chat';

describe('Chat', () => {
  it('matches snapshot', () => {
    const mockChat = {
      members: {
        3: 'person1',
        4: 'person2'
      },
      timeStamp: 34
    }
    const wrapper = shallow(<Chat chat={mockChat}/>);
    expect(wrapper).toMatchSnapshot();
  })
})