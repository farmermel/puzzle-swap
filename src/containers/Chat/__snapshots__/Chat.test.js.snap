// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chat
    chat={
        Object {
            "members": Object {
              "3": "person1",
              "4": "person2",
            },
            "timeStamp": 34,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to={
                Object {
                        "pathname": "/messages/undefined",
                      }
        }
>
        <h3>
                Members: 
                <span>
                        person1
                        person2
                </span>
        </h3>
        <h2>
                34
        </h2>
        <p>
                I'm a chat blurb
        </p>
        <p>
                hopefully one day you can click on me and I'll also print a bunch of messages
        </p>
</Link>,
      "className": "chat",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h3>
            Members: 
            <span>
                        person1
                        person2
            </span>
</h3>,
          <h2>
            34
</h2>,
          <p>
            I'm a chat blurb
</p>,
          <p>
            hopefully one day you can click on me and I'll also print a bunch of messages
</p>,
        ],
        "replace": false,
        "to": Object {
          "pathname": "/messages/undefined",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Members: ",
              <span>
                person1
                person2
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Members: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "person1",
                  "person2",
                ],
              },
              "ref": null,
              "rendered": Array [
                "person1",
                "person2",
              ],
              "type": "span",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 34,
          },
          "ref": null,
          "rendered": 34,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "I'm a chat blurb",
          },
          "ref": null,
          "rendered": "I'm a chat blurb",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hopefully one day you can click on me and I'll also print a bunch of messages",
          },
          "ref": null,
          "rendered": "hopefully one day you can click on me and I'll also print a bunch of messages",
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to={
                    Object {
                              "pathname": "/messages/undefined",
                            }
          }
>
          <h3>
                    Members: 
                    <span>
                              person1
                              person2
                    </span>
          </h3>
          <h2>
                    34
          </h2>
          <p>
                    I'm a chat blurb
          </p>
          <p>
                    hopefully one day you can click on me and I'll also print a bunch of messages
          </p>
</Link>,
        "className": "chat",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h3>
              Members: 
              <span>
                            person1
                            person2
              </span>
</h3>,
            <h2>
              34
</h2>,
            <p>
              I'm a chat blurb
</p>,
            <p>
              hopefully one day you can click on me and I'll also print a bunch of messages
</p>,
          ],
          "replace": false,
          "to": Object {
            "pathname": "/messages/undefined",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Members: ",
                <span>
                  person1
                  person2
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Members: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "person1",
                    "person2",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "person1",
                  "person2",
                ],
                "type": "span",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 34,
            },
            "ref": null,
            "rendered": 34,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "I'm a chat blurb",
            },
            "ref": null,
            "rendered": "I'm a chat blurb",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "hopefully one day you can click on me and I'll also print a bunch of messages",
            },
            "ref": null,
            "rendered": "hopefully one day you can click on me and I'll also print a bunch of messages",
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
